### PR TITLE
Identifier generator

### PR DESCRIPTION
**Pull request by Kathrin-Huber**
Making the pull request in the Name of Beatrycze, because she was only waiting for my Review.

Module is added with functionality. It's tested.

### CODE REVIEW

File: Kitodo-IdentifierGenerator/pom.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (henning-gerhardt) [inline Kitodo-IdentifierGenerator/pom.xml]: Missing license header.

File: Kitodo-IdentifierGenerator/pom.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            </resource>
+        </resources>
+        <testResources>
+            <testResource>
+                <directory>src/test/resources</directory>
+                <filtering>true</filtering>
+            </testResource>
+        </testResources>
+    </build>
+</project>
```
Comment (henning-gerhardt) [inline Kitodo-IdentifierGenerator/pom.xml]: Missing new line at end of file.

File: Kitodo-IdentifierGenerator/src/test/java/org/kitodo/identifiergenerator/DNBURNIdentifierGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for PersistentIdentifier.
+ */
+public class DNBURNIdentifierGeneratorTest {
+
+    @Test
+    public void shouldGenerateUnifiedResourceName() throws Exception {
```
Comment (henning-gerhardt) [inline Kitodo-IdentifierGenerator/src/test/java/org/kitodo/identifiergenerator/DNBURNIdentifierGeneratorTest.java]: It is possible to add more tests which are proving that this used algorithm is correct? At least different values for all four parameters should be used.

**Comment by henning-gerhardt** [comment]
- Travis run was not successful:
`[ERROR] Failed to execute goal org.flywaydb:flyway-maven-plugin:4.0.3:baseline (default) on project kitodo-identifiergenerator: org.flywaydb.core.api.FlywayException: Unable to read config file: /home/travis/build/kitodo/kitodo-production/Kitodo-IdentifierGenerator/Kitodo-DataManagement/src/main/resources/db/config/flyway.properties`
 **This must be fixed.**

- Findbugs reports for new module this message:
`[INFO] Integral division result cast to double or float in org.kitodo.identifiergenerator.DNBURNIdentifierGenerator.getCheckDigit(String) [org.kitodo.identifiergenerator.DNBURNIdentifierGenerator] At DNBURNIdentifierGenerator.java:[line 117] ICAST_IDIV_CAST_TO_DOUBLE`
 This should be fixed.

File: Kitodo-IdentifierGenerator/src/main/java/org/kitodo/identifiergenerator/DNBURNIdentifierGenerator.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+package org.kitodo.identifiergenerator;
+
+import java.util.HashMap;
+
+import org.kitodo.api.identifiergenerator.URNGeneratorInterface;
+
+/**
+ * Class for generation and registration DBNURN identifier.
+ */
+public class DNBURNIdentifierGenerator implements URNGeneratorInterface {
```
Comment (henning-gerhardt) [inline Kitodo-IdentifierGenerator/src/main/java/org/kitodo/identifiergenerator/DNBURNIdentifierGenerator.java]: Checkstyle violation on class name: to many capital letters (more then 5)

File: Kitodo-IdentifierGenerator/src/test/java/org/kitodo/identifiergenerator/DNBURNIdentifierGeneratorTest.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+package org.kitodo.identifiergenerator;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests for PersistentIdentifier.
+ */
+public class DNBURNIdentifierGeneratorTest {
```
Comment (henning-gerhardt) [inline Kitodo-IdentifierGenerator/src/test/java/org/kitodo/identifiergenerator/DNBURNIdentifierGeneratorTest.java]: Checkstyle violation on class name: to many capital letters (more then 5)

File: Kitodo-IdentifierGenerator/src/test/java/org/kitodo/identifiergenerator/DNBURNIdentifierGeneratorTest.java (hunk 3 above)
Comment (Beacze) [inline Kitodo-IdentifierGenerator/src/test/java/org/kitodo/identifiergenerator/DNBURNIdentifierGeneratorTest.java]: I see problem for other libraries.
SLUB adds to numbers 'id' but for e.g. UB Mannheim not (https://digi.bib.uni-mannheim.de/seitenansicht_sammlung/dlf/874/1/ : urn:nbn:de:bsz:180-digad-2807).

File: Kitodo-IdentifierGenerator/pom.xml (hunk 1 above)
Comment (Beacze) [inline Kitodo-IdentifierGenerator/pom.xml]: done

File: Kitodo-IdentifierGenerator/pom.xml (hunk 2 above)
Comment (Beacze) [inline Kitodo-IdentifierGenerator/pom.xml]: done

File: Kitodo-IdentifierGenerator/src/main/java/org/kitodo/identifiergenerator/DNBURNIdentifierGenerator.java (hunk 4 above)
Comment (Beacze) [inline Kitodo-IdentifierGenerator/src/main/java/org/kitodo/identifiergenerator/DNBURNIdentifierGenerator.java]: done

File: Kitodo-IdentifierGenerator/src/test/java/org/kitodo/identifiergenerator/DNBURNIdentifierGeneratorTest.java (hunk 5 above)
Comment (Beacze) [inline Kitodo-IdentifierGenerator/src/test/java/org/kitodo/identifiergenerator/DNBURNIdentifierGeneratorTest.java]: done

File: Kitodo-PersistentIdentifier/pom.xml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <sourceDirectory>src/main/java</sourceDirectory>
```
Comment (Kathrin-Huber) [inline Kitodo-PersistentIdentifier/pom.xml]: What do you need the copying for?

Comment (Beacze) [inline Kitodo-PersistentIdentifier/pom.xml]: What do you mean?

File: Kitodo-PersistentIdentifier/src/main/java/org/kitodo/persistentidentifier/URNPersistentIdentifier.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+     * @param subNamespace
+     *            the sub namespace.
+     * @param identifier
+     *            the identifier of the specific object to which the URN points.
+     * @return generated URN
+     */
+    @Override
+    public String generateUnifiedResourceName(String namespace, String libraryIdentifier, String subNamespace,
+            String identifier) {
+        String urn = namespace + ':' + libraryIdentifier + '-' + subNamespace + "-id" + identifier;
```
Comment (henning-gerhardt) [inline Kitodo-PersistentIdentifier/src/main/java/org/kitodo/persistentidentifier/URNPersistentIdentifier.java]: `id` is not part of `identifier`. For example testing with `urn:nbn:de:bsz:14-qucosa-22579` results in wrong `urn:nbn:de:bsz:14-qucosa-id225798`. Correct value must be `urn:nbn:de:bsz:14-qucosa-225799`. I can provide some more examples if you need them.

File: Kitodo-PersistentIdentifier/src/test/java/org/kitodo/persistentidentifier/URNPersistentIdentifierTest.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Tests for PersistentIdentifier.
+ */
+public class URNPersistentIdentifierTest {
+
+    @Test
+    public void shouldGenerateUnifiedResourceName() throws Exception {
+        URNPersistentIdentifier urnPersistentIdentifier = new URNPersistentIdentifier();
+
+        String urn = urnPersistentIdentifier.generateUnifiedResourceName("bsz", "14", "db", "378704842");
```
Comment (henning-gerhardt) [inline Kitodo-PersistentIdentifier/src/test/java/org/kitodo/persistentidentifier/URNPersistentIdentifierTest.java]: Missing `id` in last parameter as `id` is part of this used identifier value! This must be corrected for other tests too.

File: Kitodo-PersistentIdentifier/src/main/java/org/kitodo/persistentidentifier/URNPersistentIdentifier.java (hunk 7 above)
Comment (Beacze) [inline Kitodo-PersistentIdentifier/src/main/java/org/kitodo/persistentidentifier/URNPersistentIdentifier.java]: done

File: Kitodo-PersistentIdentifier/src/test/java/org/kitodo/persistentidentifier/URNPersistentIdentifierTest.java (hunk 8 above)
Comment (Beacze) [inline Kitodo-PersistentIdentifier/src/test/java/org/kitodo/persistentidentifier/URNPersistentIdentifierTest.java]: removed

**Pull request merged**